### PR TITLE
Feature/amazon s3 support for images

### PR DESCRIPTION
**Pull request by HansHappyHorizon**
# Describe your changes

Added support for Amazon S3 for file uploads.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How was this tested?

Used a demo account to try uploading, downloading and deleting images using Amazon S3.

# Checklist before requesting a review
- [x] I have reviewed and tested my changes
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I selected `develop` as the base branch and not `main`, or the pull request is a hotfix that needs to be done directly on `main`
- [x] I double checked all my changes and they contain no temporary test code, no code that is commented out and no changes that are not part of this branch
- [ ] I added new unit tests for my changes if applicable

# Related pull requests

N/A

# Jira issue key

CON-672

### CODE REVIEW

File: Api/Modules/Files/Services/FilesService.cs [hunk 1]
```diff
@@ -544,38 +558,60 @@ private async Task<ServiceResult<FileModel>> SaveToDatabaseAsync(ClaimsIdentity
...
-            if (ftp.Mode == FtpModes.Sftp)
+            if (ftpSettings.Any())
             {
-                throw new NotImplementedException("SFTP is not yet supported");
-                /*using var client = new SftpClient(ftp.Host, ftp.Username, ftp.Password);
+                var ftp = ftpSettings.First();
+                if (ftp.Mode == FtpModes.Sftp)
+                {
+                    throw new NotImplementedException("SFTP is not yet supported");
+                    /*using var client = new SftpClient(ftp.Host, ftp.Username, ftp.Password);
```
Comment (MikeVanMourik) [inline Api/Modules/Files/Services/FilesService.cs:573]: Is this needed?

**Review by MikeVanMourik — CHANGES_REQUESTED** [review]
One small thing, do we need that commented code in there or not?

File: Api/Modules/Files/Services/FilesService.cs (hunk 1 above)
Comment (HansHappyHorizon) [reply Api/Modules/Files/Services/FilesService.cs:573]: Not sure, I didn't put it there. I imagine we'll want SFTP support at some point, but I don't know if the code is someone's TODO or if it's too old and should be removed.

Comment (MikeVanMourik) [reply Api/Modules/Files/Services/FilesService.cs:573]: alright then ill let it be :)

**Pull request merged**